pkg/apiserver/handlers.go

```diff
@@ -188,11 +190,11 @@ func (t *timeoutHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	done := make(chan struct{}, 1)
+	done := make(chan struct{})
 	tw := newTimeoutWriter(w)
 	go func() {
 		t.handler.ServeHTTP(tw, r)
```

Comment:
(not sure if relevant, but we now crash for real when there's a panic, so
if you don't want to bring down the whole process you'll need to intercept
it somewhere.)

On Mon, Aug 8, 2016 at 3:31 PM, Xiang Li <notifications@github.com> wrote:

> In pkg/apiserver/handlers.go
> <https://github.com/kubernetes/kubernetes/pull/29594#discussion_r73968141>
> :
>
> > @@ -188,11 +190,11 @@ func (t *timeoutHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
> >  		return
> >  	}
> >
> > -	done := make(chan struct{}, 1)
> > +	done := make(chan struct{})
> >  	tw := newTimeoutWriter(w)
> >  	go func() {
> >  		t.handler.ServeHTTP(tw, r)
>
> Will this bring down the server? I thought server was recovered from
> panic. Also the log in the original issue showed that the API server
> recovered from the panic. We just did not handle the request correctly.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/29594/files/b9129837e9db2272de6f73ba88552451ae1438de#r73968141>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAnglkdhiOpVhhS3g754tu4FbvYiujbHks5qd65XgaJpZM4JUwp4>
> .
>